### PR TITLE
gradle 7.0 support

### PR DESCRIPTION
**Pull request by AaronDsilva97**
Added support for gradle 7.0 and above (android specific)

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -107,8 +107,8 @@ afterEvaluate { project ->
 
     android.libraryVariants.all { variant ->
         def name = variant.name.capitalize()
-        task "jar${name}"(type: Jar, dependsOn: variant.javaCompile) {
-            from variant.javaCompile.destinationDir
+        task "jar${name}"(type: Jar, dependsOn: variant.javaCompileProvider) {
+            from variant.javaCompileProvider.get().destinationDir
```
Comment (kayzhang) [inline android/build.gradle:99]: Not quite familiar with react native. But when I apply this commit to my project, got the below error.

```
> Configure project :react-native-securerandom
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

FAILURE: Build failed with an exception.

* Where:
Build file '/Volumes/workplace/test_5930242856/src/MShopSsnapSharedComponentsAndroid/node_modules/react-native-securerandom/android/build.gradle' line: 111

* What went wrong:
A problem occurred configuring project ':react-native-securerandom'.
> Could not get unknown property 'destinationDir' for provider(task 'compileDebugJavaWithJavac', class org.gradle.api.tasks.compile.JavaCompile) of type org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider_Decorated.
```

Remove changes on line 110~111 worked for me.

**Comment by hanno-jonlay** [comment]
Just to add to this. I had to make the following additional change to get past an issue with the use of `compile` in Gradle 7:

```
task androidJavadoc(type: Javadoc) {
    ...
    classpath += files(project.getConfigurations().getByName('compile').asList())
    ...
}
```

That change was:

```
task androidJavadoc(type: Javadoc) {
    ...
    project.getConfigurations().getByName('implementation').setCanBeResolved(true)
    classpath += files(project.getConfigurations().getByName('implementation').asList())
    ...
}
```

I believe this is a fairly common change as I've seen it on other projects, too: https://github.com/iamolegga/react-native-launch-arguments/pull/34/files

**Comment by aiden-petersen** [comment]
Hi @robhogan, are you able to look into this PR when you get a chance please, lots of apps moving to gradle 7 and will have to find another library

**Comment by HwangTaehyun** [comment]
@aiden-petersen 
How about deleting all the afterEvaluate related code?
I don't know when to use this code and throws many errors in gradle 7.0

**Review by HwangTaehyun — CHANGES_REQUESTED** [review]
@aiden-petersen 
How about deleting all the afterEvaluate related code?
I don't know when to use this code and it throws many errors in gradle 7.0.
This PR also throws error in the line: 93 
```
> Configuration with name 'compile' not found.
```

**Comment by aiden-petersen** [comment]
> @aiden-petersen How about deleting all the afterEvaluate related code? I don't know when to use this code and it throws many errors in gradle 7.0. This PR also throws error in the line: 93
> 
> ```
> > Configuration with name 'compile' not found.
> ```

@HwangTaehyun Yeah agree I'm not sure what that code is used for, happy to delete `afterEvaluate` but not sure on the implications, I can look into it

**Pull request closed without merging**